src/buffer/out/textBuffer.cpp

```diff
@@ -2472,11 +2472,24 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
 // - An ID that the caller should associate with the given pattern
 const size_t TextBuffer::AddPatternRecognizer(const std::wstring_view regexString)
 {
+    // If the pattern already exists in our map, return the id associated with it
+    for (auto idPatternPair : _idsAndPatterns)
+    {
+        if (idPatternPair.second == regexString)
+        {
+            return idPatternPair.first;
+        }

```

Comment:
```suggestion
    for (const auto [_, pattern] : _idsAndPatterns)
    {
        if (pattern == regexString)
        {
            return idPatternPair.first;
        }
```

2 things here:
1. `const`
2. (this one is just plain cool) you can use the `[]` notation to automatically unpack the pair into two variables. `_` is special in that we just straight up don't care about it ;)

Reply:
Thanks for the tip! Made the change (though in this case we do want the first variable to potentially return it)